x/mongo/driver/driver.go

```diff
@@ -127,20 +142,10 @@ func (ssd SingleConnectionDeployment) SupportsRetryWrites() bool { return false
 func (ssd SingleConnectionDeployment) Kind() description.TopologyKind { return description.Single }
 
 // Connection implements the Server interface. It always returns the embedded connection.
-//
-// This method returns a Connection with a no-op Close method. This ensures that a
-// SingleConnectionDeployment can be used across multiple operation executions.
 func (ssd SingleConnectionDeployment) Connection(context.Context) (Connection, error) {
-	return nopCloserConnection{ssd.C}, nil
```

Comment:
I don't think this wrapping is necessary. There are two instances where a connection should be used for an operation and then should not be closed or returned to a pool: handshakes and heartbeats. Both of these uses already wrap the connection in `topology.initConnection`, which serves the same purpose as `nopCloserConnection`, so the extra wrapping adds nothing.

Having this wrapping also hurts us because if `initConnection` implements `Streamer`, wrapping it in `nopCloserConnection` "gets rid of" the `Streamer` implementation.